### PR TITLE
Added condex(), hdluid(), uidxrf() ordinals (137, 338, 610)

### LINKED ISSUE #299: [GALSNV] SuperNova - Module crashes on entry
**Module Information**
* Identifier: GALSNV
* Module Name: SuperNova
* Version: 1.0

**Describe the bug**
Module crashes on entry

**To Reproduce**
1. Add module to moduleConfig.json
2. Load MBBSEmu
3. Enter module

**Expected behavior**
Able to enter module

**Screenshots**
```
2020-11-16 22:34:28.1785 Error MBBSEmu.HostProcess.ExportedModules.Majorbbs.Invoke Unknown Exported Function Ordinal in MAJORBBS: 137:CONDEX
Unhandled exception. System.ArgumentOutOfRangeException: Specified argument was out of the range of valid values. (Parameter 'Unknown Exported Function Ordinal in MAJORBBS: 137:CONDEX')
   at MBBSEmu.HostProcess.ExportedModules.Majorbbs.Invoke(UInt16 ordinal, Boolean offsetsOnly)
   at MBBSEmu.HostProcess.ExecutionUnits.ExecutionUnit.ExternalFunctionDelegate(UInt16 ordinal, UInt16 functionOrdinal)
   at MBBSEmu.CPU.CpuCore.Op_Call()
   at MBBSEmu.CPU.CpuCore.Tick()
   at MBBSEmu.HostProcess.ExecutionUnits.ExecutionUnit.Execute(IntPtr16 entryPoint, UInt16 channelNumber, Boolean simulateCallFar, Boolean bypassState, Queue`1 initialStackValues, UInt16 initialStackPointer)
   at MBBSEmu.Module.MbbsModule.Execute(IntPtr16 entryPoint, UInt16 channelNumber, Boolean simulateCallFar, Boolean bypassSetState, Queue`1 initialStackValues, UInt16 initialStackPointer)
   at MBBSEmu.HostProcess.MbbsHost.WorkerThread()
   at System.Threading.ThreadHelper.ThreadStart_Context(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location where exception was thrown ---
   at System.Threading.ThreadHelper.ThreadStart()
```

**Software Information:**
- OS: Windows Server 2019 Standard
- Telnet Client: NetRunner
- MBBSEmu Build: 111220-2

### PR DESCRIPTION
**Pull request by tuday2**
Closes #299

### CODE REVIEW

**Comment by enusbaum** [comment]
Add Unit Tests and we're good to go on this one! 👍

**Comment by tuday2** [comment]
Condex() test done, ready for initial review

File: MBBSEmu/HostProcess/Structs/UidxrefStruct.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        public const int UIDSIZ = 30; // user-id size (including trailing zero)
+        public const int SPARESIZ = 6; // spare
+        
+
+        public byte[] xrfstg
+        {
+            get
+            {
+                ReadOnlySpan<byte> xrfstgSpan = Data;
+                return xrfstgSpan.Slice(0, XRFSIZ + 1).ToArray();
```
Comment (enusbaum) [inline MBBSEmu/HostProcess/Structs/UidxrefStruct.cs:22]: You can one line these getters as `get => new ReadOnlySpan<byte>(Data).Slice(xyz).ToArray();`

File: MBBSEmu/HostProcess/ExportedModules/Majorbbs.cs [hunk 2]
```diff
@@ -7500,5 +7509,55 @@ private void echsec()
...
+            var userXrefPointer = new IntPtr16(Module.Memory.GetVariablePointer("UIDXRF").Data);
+            var userXref = new UidxrefStruct(Module.Memory.GetArray(userXrefPointer, UidxrefStruct.Size));
+
+            //Look up user ID
+            var userAccount = _accountRepository.GetAccounts().ToList().FirstOrDefault(item => item.userName.Contains(searchUserName, StringComparison.CurrentCultureIgnoreCase));
+            
+            if (userAccount != null && searchUserName != "")
+            {
+                userXref.xrfstg = Encoding.ASCII.GetBytes(searchUserName);
+                userXref.userid = Encoding.ASCII.GetBytes(userAccount.userName);
```
Comment (enusbaum) [inline MBBSEmu/HostProcess/ExportedModules/Majorbbs.cs:7542]: null terminate these using `+ "\0"`

File: MBBSEmu/HostProcess/ExportedModules/Majorbbs.cs [hunk 3]
```diff
@@ -186,6 +186,7 @@ public void Dispose()
             Module.Memory.SetWord("DIGALW", 1);
             Module.Memory.AllocateVariable("_8087", sizeof(ushort));
             Module.Memory.SetWord("_8087", 3);
+            Module.Memory.AllocateVariable("UIDXRF", 0x34, true); //52 bytes for uidxrf struct
```
Comment (enusbaum) [inline MBBSEmu/HostProcess/ExportedModules/Majorbbs.cs:189]: use the const struct `Size` for the value here vs. 0x34

**Review by enusbaum — CHANGES_REQUESTED** [review]

File: MBBSEmu/HostProcess/ExportedModules/Majorbbs.cs [hunk 4]
```diff
@@ -7500,5 +7509,55 @@ private void echsec()
...
+            if (userAccount != null && searchUserName != "")
+            {
+                userXref.xrfstg = Encoding.ASCII.GetBytes(searchUserName + "\0");
+                userXref.userid = Encoding.ASCII.GetBytes(userAccount.userName + "\0");
+                Module.Memory.SetArray(userXrefPointer, userXref.Data);
+                Registers.AX = 0;
+            }
+            else
+            {
+                userXref.xrfstg = Encoding.ASCII.GetBytes(searchUserName);
```
Comment (enusbaum) [inline MBBSEmu/HostProcess/ExportedModules/Majorbbs.cs:7548]: Null terminate these as well

File: MBBSEmu/HostProcess/Structs/UidxrefStruct.cs [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    public class UidxrefStruct
+    {
+        public const int XRFSIZ = 15; // user-id search string size
+        public const int UIDSIZ = 30; // user-id size (including trailing zero)
+        public const int SPARESIZ = 6; // spare
+        
+
+        public byte[] xrfstg
+        {
+            get => new ReadOnlySpan<byte>(Data).Slice(0, XRFSIZ + 1).ToArray();
```
Comment (enusbaum) [inline MBBSEmu/HostProcess/Structs/UidxrefStruct.cs:19]: Why is this one +1?

File: MBBSEmu/HostProcess/Structs/UidxrefStruct.cs [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    /// <summary>
+    ///     User-id cross reference structure
+    ///
+    ///     MAJORBBS.H
+    /// </summary>
+    public class UidxrefStruct
+    {
+        public const int XRFSIZ = 15; // user-id search string size
+        public const int UIDSIZ = 30; // user-id size (including trailing zero)
+        public const int SPARESIZ = 6; // spare
```
Comment (enusbaum) [inline MBBSEmu/HostProcess/Structs/UidxrefStruct.cs:14]: Based on adding these up, the `Size` should be 51 -- right?

**Review by enusbaum — CHANGES_REQUESTED** [review]

File: MBBSEmu.Tests/ExportedModules/Majorbbs/condex_Tests.cs [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+using Xunit;
+
+namespace MBBSEmu.Tests.ExportedModules.Majorbbs
+{
+    public class condex_Tests : ExportedModuleTestBase
+    {
+        private const ushort CONDEX_ORDINAL = 137;
+
+        [Fact]
+        public void condex_Test()
```
Comment (paladine) [inline MBBSEmu.Tests/ExportedModules/Majorbbs/condex_Tests.cs:14]: should have a test without the CONDEX flag set as well

**Pull request merged**